compiler/src/dotty/tools/backend/jvm/LabelDefs.scala

```diff
@@ -82,68 +56,47 @@ import StdNames.nme
  * @author Dmitry Petrashko
  */
 class LabelDefs extends MiniPhase {
-  def phaseName: String = "labelDef"
+  import tpd._
 
-  val queue = new ArrayBuffer[Tree]()
-  val beingAppended = new mutable.HashSet[Symbol]()
-  var labelLevel = 0
+  def phaseName: String = "labelDef"
 
-  override def transformDefDef(tree: tpd.DefDef)(implicit ctx: Context): tpd.Tree = {
-    if (tree.symbol is Flags.Label) tree
+  override def transformDefDef(tree: DefDef)(implicit ctx: Context): Tree = {
+    if (tree.symbol is Label) tree
     else {
-      collectLabelDefs.clear
-      val newRhs = collectLabelDefs.transform(tree.rhs)
-      var labelDefs = collectLabelDefs.labelDefs
+      val labelDefs = collectLabelDefs(tree.rhs)
 
       def putLabelDefsNearCallees = new TreeMap() {
```

Comment:
I guess this can be a `val`

Reply:
Yes, but I added a small optimization to not create the transformer if `labelDefs` is empty. Therefore it will be kept as a `def`.